src/lib/libdll/dlldefs.h

```diff
@@ -0,0 +1,170 @@
+*               This software is part of the ast package               *
+*          Copyright (c) 1997-2018 AT&T Intellectual Property          *
+*                      and is licensed under the                       *
+*                 Eclipse Public License, Version 1.0                  *
+*                    by AT&T Intellectual Property                     *
+*                                                                      *
+*                A copy of the License is available at                 *
+*          http://www.eclipse.org/org/documents/epl-v10.html           *
+*         (with md5 checksum b35adb5213ca9657e911e9befb180842)         *
+*                                                                      *
+*              Information and Software Systems Research               *
+*                            AT&T Research                             *
+*                           Florham Park NJ                            *
+*                                                                      *
+*               Glenn Fowler <glenn.s.fowler@gmail.com>                *
+*                                                                      *
+***********************************************************************/
+
+/* : : generated by proto : : */
+/* : : generated from /home/situ/test/ast/src/lib/libdll/features/dll by iffe version 2013-11-14 : : */
+                  
+#ifndef _def_dll_dll
+#if !defined(__PROTO__)
+#  if defined(__STDC__) || defined(__cplusplus) || defined(_proto) || defined(c_plusplus)
+#    if defined(__cplusplus)
+#      define __LINKAGE__	"C"
+#    else
+#      define __LINKAGE__
+#    endif
+#    define __STDARG__
+#    define __PROTO__(x)	x
+#    define __OTORP__(x)
+#    define __PARAM__(n,o)	n
+#    if !defined(__STDC__) && !defined(__cplusplus)
+#      if !defined(c_plusplus)
+#      	define const
+#      endif
+#      define signed
+#      define void		int
+#      define volatile
+#      define __V_		char
+#    else
+#      define __V_		void
+#    endif
+#  else
+#    define __PROTO__(x)	()
+#    define __OTORP__(x)	x
+#    define __PARAM__(n,o)	o
+#    define __LINKAGE__
+#    define __V_		char
+#    define const
+#    define signed
+#    define void		int
+#    define volatile
+#  endif
+#  define __MANGLE__	__LINKAGE__
+#  if defined(__cplusplus) || defined(c_plusplus)
+#    define __VARARG__	...
+#  else
+#    define __VARARG__
+#  endif
+#  if defined(__STDARG__)
+#    define __VA_START__(p,a)	va_start(p,a)
+#  else
+#    define __VA_START__(p,a)	va_start(p)
+#  endif
+#  if !defined(__INLINE__)
+#    if defined(__cplusplus)
+#      define __INLINE__	extern __MANGLE__ inline
+#    else
+#      if defined(_WIN32) && !defined(__GNUC__)
+#      	define __INLINE__	__inline
+#      endif
+#    endif
+#  endif
+#endif
+#if !defined(__LINKAGE__)
+#define __LINKAGE__		/* 2004-08-11 transition */
+#endif
+
+#define _def_dll_dll	1
+#define _sys_types	1	/* #include <sys/types.h> ok */
```

Comment:
All of these defines need to be meson feature tests rather than being hardcoded here.